### DISCUSSION TITLE
Using output from a run command in a future step

### DISCUSSION THREAD

**Discussion by nodesocket** [community/community#26139] [Actions]
<p>I have a complex command <code>./dockebuild.sh</code> that runs a bunch of things in a bash script, and at the end outputs the built Docker image tag.</p>
<pre><code>jobs:
  ecs:
    name: ecs
    runs-on: ubuntu-latest
    steps:
        - name: Checkout
          uses: actions/checkout@v1
        - name: Docker build
          run: ./dockerbuild.sh
        - name: Render ECS task definition
          uses: aws-actions/amazon-ecs-render-task-definition@v1
          with:
            task-definition: task-definition.json
            container-name: stage-web
            image: &lt;a value from the output of ./dockerbuild.sh here&gt;
</code></pre>
<p>How can I grab just what I need from the output of <code>./dockerbuild.sh</code> and use it in <strong>Render ECS task definition?  </strong> <em>Note: there is tons of output from <code>./dockebuild.sh</code>but I need to search for a specific regular expression and use that.</em></p>

**Comment by nodesocket** [comment]
<p>I just tried:</p>
<pre><code>export GA_DOCKER_IMAGE="repo-here:tag-here"
</code></pre>
<p>Inside of the script <code>dockerbuild.sh</code> and updating the run command to:</p>
<pre><code>. dockerbuild.sh
</code></pre>
<p>Then using the exported ENVAR like:</p>
<pre><code>- name: Render new ECS task definition
  uses: aws-actions/amazon-ecs-render-task-definition@v1
  with:
    task-definition: task-definition.json
    container-name: stage-web
    image: ${{ env.GA_DOCKER_IMAGE }}
</code></pre>
<p>But ${{ env.GA_DOCKER_IMAGE }} is empty. Any ideas?</p>

**Comment by rbellamy** [comment]
<p>Try:</p>
<pre><code>echo "::set-env name=GA_DOCKER_IMAGE::repo-here:tag-here"
</code></pre>
<p><a href="https://help.github.com/en/actions/reference/development-tools-for-github-actions#set-an-environment-variable-set-env" rel="nofollow noopener">https://help.github.com/en/actions/reference/development-tools-for-github-actions#set-an-environment-variable-set-env</a></p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/nodesocket">@nodesocket</a> ,</p>
<p>You need to filter the output result of the dockebuild.sh file, eg: use ‘grep’.</p>
<p>For instance, two outputs in sh file:</p>
<pre><code>echo tag1=v1
echo tag2=v2
</code></pre>
<p>Then in workflow yaml file:</p>
<pre><code>- name: get output value
        id: getimage
        run: |
          echo "::set-output name=tag::$(sh dockerbuild.sh | grep tag2)"
      - name: check the value
        run: |
          echo ${${{ steps.getimage.outputs.tag }}} # this will show 'v2', is the expected tag.
</code></pre>
<p>Or directly use ‘set-env’ in dockebuild.sh file:</p>
<pre><code>echo "::set-env name=GA_DOCKER_IMAGE::repo-here:tag2"
</code></pre>
<p>then you can invoke it in yaml file with ‘$GA_DOCKER_IMAGE’ and used as action parameters.</p>
<pre><code>- name: get the value from env
        run: |
          echo $GA_DOCKER_IMAGE
</code></pre>
<p>Hope it helps!</p>

**Comment by nodesocket** [comment]
<p>Thanks <a class="mention" href="/weide-zhou">@weide-zhou</a> and <a class="mention" href="/rbellamy">@rbellamy</a>:</p>
<p>I will try this. What is this syntax? Is it specific to running inside of a GitHub action bash script?</p>
<pre><code>echo "::set-env name=GA_DOCKER_IMAGE::repo-here:tag2"
</code></pre>

**Comment by weide-zhou** [comment]
<p>Please check the <a href="https://help.github.com/en/actions/reference/development-tools-for-github-actions#set-an-environment-variable-set-env" rel="nofollow noopener">link</a> for more details about ‘set-env’. Actually it’s a wrapped github action function which described <a href="https://github.com/actions/toolkit/blob/master/docs/commands.md#set-an-environment-variable" rel="nofollow noopener">here</a>, it’s only used in Github actions. Thanks.</p>
